Explorer/Assets/DCL/Multiplayer/Movement/Systems/RemotePlayerAnimationSystem.cs

```diff
@@ -0,0 +1,156 @@
+                UpdateAnimations(view, ref anim, remotePlayerMovement.PastMessage.animState, remotePlayerMovement.PastMessage.isStunned);
+            }
+
+            if (intComp.Enabled)
+                InterpolateAnimations(view, ref anim, intComp);
+            else if (extComp.Enabled)
+                ExtrapolateAnimations(view, ref anim, extComp.Time, extComp.TotalMoveDuration, settings.LinearTime);
+        }
+
+        private static void UpdateAnimations(IAvatarView view, ref CharacterAnimationComponent animationComponent, in AnimationStates animState, bool isStunned)
+        {
+            if (animationComponent.States.Equals(animState))
+                return;
+
+            UpdateAnimatorBlends(view, animState);
+
+            if ((animationComponent.States.IsGrounded && !animState.IsGrounded) || (!animationComponent.States.IsJumping && animState.IsJumping))
+                view.SetAnimatorTrigger(AnimationHashes.JUMP);
+
+            view.SetAnimatorBool(AnimationHashes.STUNNED, isStunned);
+            view.SetAnimatorBool(AnimationHashes.GROUNDED, animState.IsGrounded);
+            view.SetAnimatorBool(AnimationHashes.JUMPING, animState.IsJumping);
+            view.SetAnimatorBool(AnimationHashes.FALLING, animState.IsFalling);
+            view.SetAnimatorBool(AnimationHashes.LONG_JUMP, animState.IsLongJump);
+            view.SetAnimatorBool(AnimationHashes.LONG_FALL, animState.IsLongFall);
+
+            animationComponent.States = animState;
+        }
+
+        private static void UpdateAnimatorBlends(IAvatarView view, in AnimationStates animStates)
+        {
+            if (!animStates.IsGrounded)
+                return;
+
+            view.SetAnimatorFloat(AnimationHashes.MOVEMENT_BLEND, animStates.MovementBlendValue > BLEND_EPSILON ? animStates.MovementBlendValue : 0f);
+            view.SetAnimatorFloat(AnimationHashes.SLIDE_BLEND, animStates.SlideBlendValue > BLEND_EPSILON ? animStates.SlideBlendValue : 0f);
+        }
+
+        private static void InterpolateAnimations(IAvatarView view, ref CharacterAnimationComponent anim, in InterpolationComponent intComp)
+        {
+            if (!anim.States.IsJumping && intComp.End.animState.IsJumping && Mathf.Abs(intComp.Start.position.y - intComp.End.position.y) > JUMP_EPSILON)
+                AnimateFutureJump(view, ref anim, intComp.End.animState);
+
+            AnimationStates startAnimStates = intComp.Start.animState;
+            AnimationStates endAnimStates = intComp.End.animState;
+
+            bool bothPointBlendsAreZero = startAnimStates.MovementBlendValue < BLEND_EPSILON && endAnimStates.MovementBlendValue < BLEND_EPSILON
+                                                                                             && startAnimStates.SlideBlendValue < BLEND_EPSILON && endAnimStates.SlideBlendValue < BLEND_EPSILON;
+
+            if (bothPointBlendsAreZero && Vector3.SqrMagnitude(intComp.Start.position - intComp.End.position) > MOVEMENT_EPSILON)
```

Comment:
You use `MOVEMENT_EPSILON` as a squared value, I wonder if it is intentional because `sqrt(0.1) = 0.316228`. I believe the value should be `0.01`